### PR TITLE
[google_maps_flutter_web] Fix InfoWindows and getLatLng.

### PR DESCRIPTION
**Pull request by ditman**
## Description

This PR addresses:

* InfoWindows don't render their 'snippet' when loaded as Initial Geometry (passed to the GoogleMap Widget directly)
* InfoWindows don't propagate click events (turns out, the JS SDK doesn't offer that event, so we need to fake it)
* GoogleMapController.getLatLng() returns a bogus LatLng (turns out, we weren't using the right formula in that method!).

In order to synthesize the click event on the InfoWindow contents, this PR migrates the InfoWindow contents from being an HTML string to being something generated with dart:html. I had to update a previous test slightly to account for this difference.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/67854
Fixes https://github.com/flutter/flutter/issues/67289
Fixes https://github.com/flutter/flutter/issues/67606

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter_web/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+## 0.1.0+5
+
+* Update `package:google_maps` to `^3.4.5`.
+* Fix GoogleMapController.getLatLng(). [Issue](https://github.com/flutter/flutter/issues/67606).
```
Comment (blasten) [inline packages/google_maps_flutter/google_maps_flutter_web/CHANGELOG.md:4]:
```suggestion
* Fix `GoogleMapController.getLatLng()`. [Issue](https://github.com/flutter/flutter/issues/67606).
```

File: packages/google_maps_flutter/google_maps_flutter_web/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
...
+* Make InfoWindow contents clickable so onTap works as advertised. [Issue](https://github.com/flutter/flutter/issues/67289).
```
Comment (blasten) [inline packages/google_maps_flutter/google_maps_flutter_web/CHANGELOG.md:5]:
```suggestion
* Make `InfoWindow` contents clickable so onTap works as advertised. [Issue](https://github.com/flutter/flutter/issues/67289).
```

File: packages/google_maps_flutter/google_maps_flutter_web/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,10 @@
...
+* Fix InfoWindow snippets when converting initial markers. [Issue](https://github.com/flutter/flutter/issues/67854).
```
Comment (blasten) [inline packages/google_maps_flutter/google_maps_flutter_web/CHANGELOG.md:6]:
```suggestion
* Fix `InfoWindow` snippets when converting initial markers. [Issue](https://github.com/flutter/flutter/issues/67854).
```

File: packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart [hunk 4]
```diff
@@ -270,14 +270,15 @@ Set<Marker> _rawOptionsToInitialMarkers(Map<String, dynamic> rawOptions) {
...
-        if (rawMarker['infoWindow'] != null || rawMarker['snippet'] != null) {
-          String title = rawMarker['infoWindow'] != null
-              ? rawMarker['infoWindow']['title']
-              : null;
-          infoWindow = InfoWindow(
-            title: title ?? '',
-            snippet: rawMarker['snippet'] ?? '',
-          );
+        if (rawMarker['infoWindow'] != null) {
+          final title = rawMarker['infoWindow']['title'];
```
Comment (cyanglaz) [inline packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart:274]: nits: We prefer to explicitly declare types: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-var

So maybe: `final String title = ..`

ditto several other places in the PR.

**Review by cyanglaz — APPROVED** [review]
LGTM % nits

File: packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart [hunk 5]
```diff
@@ -378,13 +379,30 @@ gmaps.InfoWindowOptions _infoWindowOptionsFromMarker(Marker marker) {
...
+    final snippet = DivElement()
+      ..className = 'infowindow-snippet'
+      ..setInnerHtml(
+        sanitizeHtml(marker.infoWindow.snippet),
+        treeSanitizer: NodeTreeSanitizer.trusted,
+      );
+    container.children.add(snippet);
+  }
+
+  assert(container.children.isNotEmpty);
```
Comment (blasten) [inline packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart:402]: is this case possible outside of the test environment?

File: packages/google_maps_flutter/google_maps_flutter_web/CHANGELOG.md (hunk 2 above)
Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter_web/CHANGELOG.md:5]: Thanks for sending suggestions! I'll commit all of these alongside the other fixes so I don't trigger CI multiple times!

File: packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart (hunk 4 above)
Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart:274]: Should we enable the `always_specify_types` analysis option, so this gets enforced automatically? I think it's enabled in Flutter as well (but this is *forbidden* in g3, btw)

Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart:274]: https://dart-lang.github.io/linter/lints/always_specify_types.html

File: packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart (hunk 5 above)
Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart:402]: Nope, I'll remove the assert.

File: packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart (hunk 4 above)
Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart:274]: I've tried enabling this in the repo and... it seems we haven't enforced that rule in a while :)

**Pull request merged**